packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+/// that creates an instance of that class.
+///
+/// The [WidgetsApp] class creates a `LocalizedResources` widget so most apps
+/// will not need to create one. The widget app's `LocalizedResources` delegate can
+/// be initialized with [WidgetsApp.localizedResourcesDelegate]. The [MaterialApp]
+/// class also provides a `localizedResourcesDelegate` parameter that's just
+/// pass along to the [WidgetsApp].
+///
+/// Apps should retrieve localized resources with the [LocalizedResourceData]
+/// returned by `LocalizedResources.of(context)`. This is conventionally done
+/// by a static `.of` method on the class that defines the app's localized
+/// resources.
+///
+/// For example, using the `MyLocalizedResouces` class defined below, one would
+/// lookup a localized title string like this:
+/// ```dart
+/// MyLocalizedResources.of(context).title()
+/// ```
+/// If the `LocalizedResources` were to be rebuilt with a new locale then
+/// the widget subtree that corresponds to [BuildContext] `context` would
+/// be rebuilt after the corresponding resources had been loaded.
+///
+/// This class is effectively an [InheritedWidget]. If it's rebuilt with
+/// a new `locale` or if its `delegate.updateShouldNotify` returns true,
+/// widgets that have created a dependency by calling
+/// `LocalizedResources.of(context)` will be rebuilt after the resources
+/// for the new locale have been loaded.
+///
+/// ## Sample code
+///
+/// This following class is defined in terms of the
+/// [Dart `intl` package](https://github.com/dart-lang/intl). Using the `intl`
+/// package isn't required.
+///
+/// ```dart
+/// class MyLocalizedResources {
+///   MyLocalizedResources(this.locale);
+///
+///   final Locale locale;
+///
+///   static Future<MyLocalizedResources> load(Locale locale) {
+///     return initializeMessages(locale.toString())
+///       .then((Null _) {
+///         return new Future<MyLocalizedResources>.value(new MyLocalizedResources(locale));
+///       });
+///   }
+///
+///   static MyLocalizedResources of(BuildContext context) {
+///     return LocalizedResources.of(context).resourcesFor<MyLocalizedResources>(MyLocalizedResources);
+///   }
+///
+///   String title() => Intl.message('<title>', name: 'title', locale: locale.toString());
+///   // ... more Intl.message() methods like title()
+/// }
+/// ```
+/// A class based on the `intl` package imports a generated message catalog that provides
+/// the `initializeMessages()` function and the per-locale backing store for `Intl.message()`.
+/// The message catalog is produced by an `intl` tool that analyzes the source code for
+/// classes that contain `Intl.message()` calls. In this case that would just be the
+/// `MyLocalizedResources` class.
+///
+/// One could choose another approach for loading localized resources and looking them up while
+/// still conforming to the structure of this example.
+class LocalizedResources extends StatefulWidget {
```

Comment:
Localizations (and LocalizationsDelegate, DefaultLocalizationsDelegate, etc) is an improvement over the current names.